### ISSUE TITLE
Treatment of Hashable in invariant type argument

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#3258]
Consider this example (checked in latest Pylance in VSCode, as always):

```python
from typing import Hashable


def f(x: list[Hashable]):
    x.append("x")


def g(x: list[object]):
    pass


l: list[int] = [1]
f(l)  # no error
g(l)  # error (as expected)
```

`f(l)` should be an error, because `list[]` is invariant and `f` wants a `list[Hashable]`, not `list[int]`. It errors correctly when the callee is annotated as `list[object]` instead.

Context: https://github.com/python/mypy/issues/12464#issuecomment-1079759274

**Comment by erictraut** [comment]
Wow, this was a pretty major hole. Thanks for reporting.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.233, which I just published. It will also be included in the next release of pylance.
